### PR TITLE
Make `Generate_Wiki_Content` a metatask

### LINKED ISSUE #135: Make `Generate_Wiki_Content` a metatask that runs other build tasks
Suggest making `Generate_Wiki_Content` a metatask and the current `Generate_Wiki_Content` is renamed to `Generate_Markdown_For_DSC_Resources` (or something similar). 

The new metatask `Generate_Wiki_Content` will run all the tasks that is needed to generate the wiki content:

- `Generate_Markdown_For_DSC_Resources`
- `Generate_Markdown_For_Public_Commands`
- `Generate_External_Help_File_For_Public_Commands`
- `Clean_Markdown_Of_Public_Commands`

### PR DESCRIPTION
**Pull request by dan-hughes**
#### Pull Request (PR) description

Change `Generate_Wiki_Content` to a metatask, split existing `Generate_Wiki_Content` into individual subtasks.
Update tests for new tasks.

#### This Pull Request (PR) fixes the following issues

- Fixes #135 

#### Task list

- [x] Added an entry to the change log under the Unreleased section of the file CHANGELOG.md.
      Entry should say what was changed and how that affects users (if applicable), and
      reference the issue being resolved (if applicable).
- [ ] Documentation added/updated in README.md.
- [ ] Comment-based help added/updated.
- [ ] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

- - -
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/DscResource.DocGenerator/145)

### CODE REVIEW

**Comment by dan-hughes** [comment]
@johlju, for review. 

I have an issue I do not understand how to resolve. The metatask does not know how to resolve the tasks, what am I doing wrong?

Changelog and Docs are incoming.

**Comment by johlju** [comment]
Hmm. I will take a look, but might not have time today, and maybe not tomorrow either. But will be back as soon as possible. 🙂

**Comment by johlju** [comment]
Building the module using this PR it seems to work. 

Running `./build.ps1 ?` shows the metatask and that it will run the other tasks. Running `./build.ps1 -Task Generate_Wiki_Content` works too. So for some reason the tests don't find it.

**Comment by johlju** [comment]
It is because in the tests we just point out one of the needed script files, e.g.

```powershell
invoke-build -Task Generate_Wiki_Content -File .\output\DscResource.DocGenerator\0.13.0\tasks\Generate_Wiki_Content.build.ps1
```

It returns the same error:
```
ERROR: Task 'Generate_Wiki_Content': Missing task 'Create_Wiki_Output_Folder'.
At C:\source\DscResource.DocGenerator\output\DscResource.DocGenerator\0.13.0\tasks\Generate_Wiki_Content.build.ps1:87 char:1
+ Task Generate_Wiki_Content Create_Wiki_Output_Folder, Copy_Source_Wik …
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Build ABORTED C:\source\DscResource.DocGenerator\output\DscResource.DocGenerator\0.13.0\tasks\Generate_Wiki_Content.build.ps1. 0 tasks, 1 errors, 0 warnings 00:00:00.0277892
Invoke-Build.ps1: Task 'Generate_Wiki_Content': Missing task 'Create_Wiki_Output_Folder'.
At C:\source\DscResource.DocGenerator\output\DscResource.DocGenerator\0.13.0\tasks\Generate_Wiki_Content.build.ps1:87 char:1
+ Task Generate_Wiki_Content Create_Wiki_Output_Folder, Copy_Source_Wik …
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by johlju** [comment]
TLDR; I think we skip testing the metatask. 🙂 

I have tried building a temp script file that dot-sources all the required task script files, but it errors out on not having the parameters we pass in the test.

```
    [-] Should run the build task without throwing 44ms
      Expected no exception to be thrown, but an exception "A parameter cannot be found that matches parameter name 'ProjectName'." was thrown from C:\source\DscResource.DocGenerator\tests\unit\tasks\Generate_Wiki_Content.build.Tests.ps1:55 char:25
          +              ) -Verbose @taskParameters
          +                         ~~~~~~~~~~~~~~~.
      56:         } | Should -Not -Throw
      at <ScriptBlock>, C:\source\DscResource.DocGenerator\tests\unit\tasks\Generate_Wiki_Content.build.Tests.ps1: line 44
```

So only way to make this work is to dynamically build a task script file with all the required tasks in it during testing, but that is not worth the effort so I suggest we skip testing the metatask. I think `Invoke-Build` assumes there should be only on script file with all tasks in it, the Sampler build script handles this that way it allows us to split up the tasks in separate files . Maybe the pipeline in this module should actually build the tasks script files into a single file in the built module. Then there would for example only be one script file that holds all parameters and they are not duplicate though out the task script files. 🤔 We would then just test one single script file in each test file. That it can be a future improvement for the project Sampler, it is a big undertaking fixing that - sounds like a "bad weather" project.  😄

**Comment by dan-hughes** [comment]
Done.

**Review by johlju — APPROVED** [review]
:lgtm:

> Reviewed 9 of 11 files at r1, 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/DscResource.DocGenerator/145#-:-NzDao0I3kCDEuTqaGxG:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dan-hughes)

**Comment by johlju** [comment]
@dan-hughes Impressive work! Awesome! Thank you! 🙇‍♂️
